### DISCUSSION TITLE
Reddit/Youtube thumbnail-like images?

### DISCUSSION THREAD

**Discussion by rieje** [emacs-elfeed/elfeed#612] [General]
Is it possible to configure elfeed to show image thumbnails of feed items next to their headlines in a fashion similar to Reddit/Youtube lists? For example, on the left column are thumbnail images and on the right column are the item headlines. I'm not talking about seeing the specific image of each item one at a time, which is different because I can't glance all the thumbnails at the same time.

This is extremely useful in for certain feeds like Youtube or Slickdeals feeds where thumbnails provide context that text-only would not otherwise allow. It is far more efficient to glance over a page of thumbnails and if I see ones that are interesting, I continue reading the item headline before viewing the item itself, as opposed to reading each headline until you realize it's not what you're interested in.

P.S. Does thumbnail support depend solely on whether the feed supports (provides) it or is it something that can be generated by the RSS reader itself?

**Comment by skeeto** [comment]
I'm guessing you're the one who asked the same thing here:

  https://redd.it/6kuu1l

I'll copy my response here, for the record:

You could customize elfeed-search-print-entry-function to your own 
function that inserts an image in the buffer, like how images appear in 
the elfeed-show buffer. There are some issues to be aware of:

* Atom has an optional logo element for feeds to act as a sort of 
favicon, but not for individual entries. There's no formal Atom metadata 
element for entry thumbnails. Generally these appear informally within 
the HTML of the content of the entry. You'd need to figure out how to 
discover these. I see YouTube uses Media RSS extension (which Yahoo! has 
abandoned) for thumbnails, though Elfeed currently doesn't capture this 
information.

* Emacs' bindings to image libraries is fragile. It's unfortunately not 
unusual for Emacs to crash when displaying an image, especially with 
GIFs and SVGs. Regularly populating an Emacs buffer with lots of images 
from the internet may have unpleasant results.

**Comment by minad** [comment]
I am a little unsure about this feature request. Therefore I converted it to a discussion. I think it might be a good fit in elfeed-tube or other extensions which specifically address multimedia feeds. On the other hand the question is what we want to support here as baseline. It may be related to the media rss feature request (#613).
